src/SparkFun_u-blox_SARA-R5_Arduino_Library.cpp

```diff
@@ -1580,6 +1616,55 @@ int8_t SARA_R5::rssi(void)
   return rssi;
 }
 
+SARA_R5_error_t SARA_R5::getExtSignalQuality(signal_quality& signal_quality)
+{
+    char *command;
+    char *response;
+    SARA_R5_error_t err;
+
+    command = sara_r5_calloc_char(strlen(SARA_R5_EXT_SIGNAL_QUALITY) + 1);
+    if (command == nullptr)
+    {
+        return SARA_R5_ERROR_OUT_OF_MEMORY;
+    }
+
+    sprintf(command, "%s", SARA_R5_EXT_SIGNAL_QUALITY);
+
+    response = sara_r5_calloc_char(minimumResponseAllocation);
+    if (response == nullptr)
+    {
+        free(command);
+        return SARA_R5_ERROR_OUT_OF_MEMORY;
+    }
+
+    err = sendCommandWithResponse(command,
+                                  SARA_R5_RESPONSE_OK_OR_ERROR, response, 10000,
+                                  minimumResponseAllocation, AT_COMMAND);
+    if (err != SARA_R5_ERROR_SUCCESS)
+    {
+        free(command);
+        free(response);
+        return SARA_R5_ERROR_ERROR;
+    }
+
+    int scanned = 0;
+    char *searchPtr = strstr(response, "+CESQ: ");
+    if (searchPtr != nullptr) {
+        scanned = sscanf(searchPtr, "+CESQ: %u,%u,%u,%u,%u,%u", &signal_quality.rxlev, &signal_quality.ber,
```

Comment:
Please restructure this bit. Search for +CESQ: first, then skip spaces, then start reading the signal quality.
This library works with other modules like the LARA and (IIRC) the LARA doesn't insert the space after the colon.
The other ```get``` methods all do the same thing - you'll see "skip spaces" comments everywhere...

Reply:
Did this.

Reply:
Thanks!